src/main/java/org/zalando/nakadi/cache/Cache.java

```diff
@@ -0,0 +1,42 @@
+package org.zalando.nakadi.cache;
+
+import java.util.function.Consumer;
+
+public interface Cache<T> {
+    /**
+     * Returns value associated with a key. If value is not found, returns null.
+     *
+     * @param key Key to get value for
+     * @return value associated with a key or null, if nothing was found.
+     */
+    T get(String key);
+
+    /**
+     * Sometimes it is important to ensure that cache will be providing refreshed version of value.
+     * This method ensures that value will be refreshed.
+     *
+     * @param key key to invalidate
+     */
+    void invalidate(String key);
+
+    /**
+     * Adds listener for event type invalidation. It is guaranteed, that any event type that was loaded into cache
```

Comment:
Your comment mentions event-type but as I understand this cache interface is for general purpose, not specifically for event-type.